components/observations/ObservationsListView.tsx

```diff
@@ -201,13 +204,17 @@ export const ObservationsListView: React.FunctionComponent<ObservationsListViewP
   );
 
   const hasPendingObservations = pendingObservationsSection.data.length > 0;
```

Comment:
Is there some gotcha here where the reference in `pendingObservationsSection` stays stable but the data underneath changes? Any reason not to depend on `pendingObservationsSection.data.length` in the dependency array and do this computation in the memo?

Reply:
We could do this in the memo. I wasn't looking too closely at the code being memoized. In the end it wouldn't really change the performance characteristics since this boolean value being changed would mean `pendingObservartionsSection` changed which would bust the memo cache dependencies.

In the end, fewer dependencies is better.

Reply:
Looks like `hasPendingObservations` is used elsewhere in the component, and memoizing that wouldn't be worth it. But we can still clean up the dependencies. `.length` is not an expensive operation.

Reply:
The hope of this change is perhaps reducing renders of `ObservationListView` which does look like it could be a expensive.